pkg/update/update.go

```diff
@@ -459,3 +471,93 @@ func updateNoIP(httpClient *http.Client, hostname, username, password, ip string
 	}
 	return "", fmt.Errorf("unknown response: %s", s)
 }
+
+func updateDnsPod(httpClient *http.Client, domain, host, token, ip string) (newIP string, err error) {
+	postValues := url.Values{}
+	postValues.Set("login_token", token)
+	postValues.Set("format", "json")
+	postValues.Set("domain", domain)
+	postValues.Set("length", "200")
+	postValues.Set("sub_domain", host)
+	postValues.Set("record_type", "A")
+	req, err := http.NewRequest(http.MethodPost, "https://dnsapi.cn/Record.List", bytes.NewBufferString(postValues.Encode()))
+	if err != nil {
+		return "", err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return "", err
+	}
+	respBody, err := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
+	if err != nil {
+		return "", err
+	}
+	type RecordResp struct {
+		Records []*struct {
+			ID    string `json:"id"`
+			Value string `json:"value"`
+			Type  string `json:"type"`
+			Name  string `json:"name"`
+			Line  string `json:"line"`
+		} `json:"records"`
+	}
+	recordResp := new(RecordResp)
+	err = json.Unmarshal(respBody, recordResp)
+	if err != nil {
+		return "", err
+	}
+	var recordID string
+	var line string
+	for _, record := range recordResp.Records {
+		if record.Type == "A" && record.Name == host {
+			recordID = record.ID
+			line = record.Line
+			if ip == record.Value {
+				return "", nil
+			}
+		}
+	}
+	if recordID == "" {
+		return "", errors.New("record not found")
```

Comment:
Please use `fmt.Errorf("record not found")` to avoid using the `errors` package on top of `fmt` (more flexible and extensible)